src/references/ui_text_entry.md

```diff
@@ -0,0 +1,114 @@
+
+<div class="mt-video" style="width:100%">
+    <video autoplay muted playsinline controls loop class="center" style="width:100%">
+        <source src="/img/UITextEntry/preview.mp4" type="video/mp4" />
+    </video>
+</div>
+
+### Use Cases
+
+There are many use cases as to why creators would want to provide a text field for users to enter text. Below lists a few ideas of how it could be used.
+
+- Customization
+
+    Users can name things (for example, a Pet) which allows for more customization.
+
+- Searching
+
+    A search feature (for example, an Inventory) allows users to find things quicker than just scrolling through long lists.
+
+- Text Adventure Games
+
+    Text-based adventure games that allow users to enter commands to move through the world.
+
+- Command Panel
+
+    Instead of using the Chat API, the text entry object could be used to capture commands.
+
+### UI Text Entry Object
+
+The **UI Text Entry** object can be added to the Hierarchy from two places.
+
+#### Core Content Window
+
+In **Core Content** under **UI Elements**, the **UI Text Entry Box** can be found and dragged into the **Hierarchy**.
+
+![!Core Content](../img/UITextEntry/core_content.png){: .center loading="lazy" }
+
+#### Create Menu
+
+When right-clicking in the **Hierarchy**, creators can filter the menu and find the **Create UI Editable Text** item.
+
+![!Create Menu](../img/UITextEntry/create_menu.png){: .center loading="lazy" }
+
+#### Properties
+
+Below are some of the properties for the **UI Text Entry Box**. Other properties not listed are similar to other controls (for example, font color, size, dock, visibility).
+
+| Property | Description |
+| -------- | ----------- |
+| `Is Hittable` | If true, the cursor will be able to interact with this object. If false, the cursor won't be able to interact with this object. |
+| `Is Interactable` | Whether text control is enabled. |
+| `Is Multiline` | The text field can span multiple lines. |
+| `Image` | The background image of the control. |
+| `Background Color` | The base color of the control. |
+| `Hovered Color` | The color when the cursor moves over the control. |
+| `Active Color` | The color when the user is editing the control. |
+| `Disabled Color` | The color when the control is disabled. |
+| `Selection Color` | The color of the text selection when highlighting all or part of the string. |
+| `Only Use Background Color` | If true, ignores Hovered, Active, and Disabled colors. |
+
+#### Behavior
+
+At runtime, a UI text entry object that **is interactable** can be clicked on with a mouse.
+
+When not being interacted with, the UI text entry object is in its background state. If the UI text entry object is empty and **Prompt** is true then the default text (using Prompt Color) is displayed. If the user has entered text or the creator has applied non-prompt text, the text is also displayed.
```

Comment:
the last sentence: "If the user has entered text or the creator has applied non-prompt text, then that text is displayed."

Reply:
Haven't used it yet, but is there a boolean property `Prompt` or is it just checking if `promptText` is not empty?

Reply:
It checks if the prompt text has a value. It can be left empty and nothing will show.